gui/src/app/Project/ProjectReducer.ts

```diff
@@ -71,8 +71,12 @@ const ProjectReducer = (s: ProjectDataModel, a: ProjectReducerAction) => {
     }
     case "commitFile": {
       const newState = { ...s };
-      if (a.filename === ProjectKnownFiles.DATAFILE) {
-        newState.meta = { ...s.meta, dataSource: undefined };
+      const newDataSource = confirmDataSourceForCommit(
+        s.meta.dataSource,
+        a.filename,
+      );
+      if (newDataSource !== s.meta.dataSource) {
```

Comment:
If they _are_ equal, then the next line is a no-op right? So we could just not check?

Reply:
Not exactly. The spread operator is creating a (shallow) copy of the object. So we'd be setting `newState.meta` to a new object with the exact same values as the original one.

In this case I don't think that matters at all, but I try to avoid it on principle because memoization checks depend on referential integrity rather than practical equivalence, so I tend to be suspicious of it.